public/components/common/restart-modal/restart-modal.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useEffect, useState } from 'react';
+import {
+  EuiButton,
+  EuiEmptyPrompt,
+  EuiEmptyPromptProps,
+  EuiOverlayMask,
+  EuiProgress,
+} from '@elastic/eui';
+import { getHttp } from '../../../kibana-services';
+import { WazuhConfig } from '../../../react-services/wazuh-config';
+import { RestartHandler } from '../../../react-services/wz-restart';
+import { getAssetURL, getThemeAssetURL } from '../../../utils/assets';
+import store from '../../../redux/store';
+
+/**
+ * Available states of the model
+ * TODO the state owner must be the Restart Service, not the modal. For this,
+ * TODO the restart service must be migrated to Typescript.
+ */
+enum RESTART_STATES {
+  ERROR = 'error',
+  RESTARTING = 'restarting',
+  SYNCING = 'syncing',
+}
+
+/**
+ * The restart modal to show feedback to the user.
+ * @param props component's props
+ * @returns components's body
+ */
+export const RestartModal = (props: { isRestarting: boolean; useDelay?: boolean }) => {
+  // Component props
+  const { isRestarting: isRestartingProp, useDelay = false } = props;
+
+  // Modal's restart state
+  const [currentState, setCurrentState] = useState(RESTART_STATES.RESTARTING);
+
+  // Restart attempt
+  const [restartAttempt, setRestartAttempt] = useState(
+    store.getState().appStateReducers.restartAttempt
+  );
```

Comment:
**nitpick**: Why are we using a global state to set the component state?

We should not use the Redux store directly. If we need a global state store in Redux, then we use the `useSelector` hook.